### PR TITLE
Liked songs rework

### LINKED ISSUE #382: only possibility to add liked songs to a beacon and a profile + modify the limit on the liked songs

### PR DESCRIPTION
**Pull request by mbouaissi**
This PR does 2 things:
1. Remove all the things related to the top songs => now we only have the liked songs and can choose from it
2. Before, we can only select one song from the 50 first liked songs => now we can select it from all the liked songs, by navigating in the selectSongScreen.

It also refactor some codes, such as the buttons,...

Closes #382

### CODE REVIEW

File: app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/ProfileViewOnlyScreen.kt [hunk 1]
```diff
@@ -15,24 +15,15 @@ import androidx.compose.ui.Modifier
...
-val mockSongLists =
-    listOf(
-        SongList(
-            ListType.TOP_SONGS,
-            listOf(Track("1", "Track 1", "Artist 1"), Track("2", "Track 2", "Artist 2"))),
-        SongList(
-            ListType.LIKED_SONGS,
-            listOf(Track("3", "Track 3", "Artist 3"), Track("4", "Track 4", "Artist 4"))))
+val mockSongLists = listOf(Track("3", "Track 3", "Artist 3"), Track("4", "Track 4", "Artist 4"))
 // TODO: modify this, because the profile.songLists is not available yet
```
Comment (IB-12) [inline app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/ProfileViewOnlyScreen.kt:27]: Has this TODO been addressed ?

**Review by IB-12 — APPROVED** [review]
LGTM, well done for this, you made the code so much cleaner. Just left a comment about a TODO you left.

File: app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/ProfileViewOnlyScreen.kt (hunk 1 above)
Comment (mbouaissi) [reply app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/ProfileViewOnlyScreen.kt:27]: not yet, but it will soon be

**Review by IB-12 — APPROVED** [review]
Still LGTM 👍

**Pull request merged**